src/Compilers/VisualBasic/Portable/Compilation/VisualBasicCompilation.vb

```diff
@@ -2654,11 +2630,18 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 i = i + 1
             Next
 
-            Dim fields = memberTypes.ZipAsArray(
-                memberNames,
-                Function(type, name) New AnonymousTypeField(name, DirectCast(type, TypeSymbol), Location.None))
+            Dim fields = ArrayBuilder(Of AnonymousTypeField).GetInstance()
+
+            For i = 0 To memberTypes.Length - 1
+                Dim type = memberTypes(i)
+                Dim name = memberNames(i)
+                Dim loc = If(memberLocations.IsDefault, Location.None, memberLocations(i))
+                Dim isReadOnly = memberIsReadOnly.IsDefault OrElse memberIsReadOnly(i)
```

Comment:
We are changing the default behavior of this API.  And that's intentional.  We want the API to produce the same symbol on the C# and VB side.  So we decided that leaving out the argument would produce an anonymous type with readonly properties for VB and C#.

Note that the IDE is the only consumer of this API (it was invented this cycle for our needs).  So it's ok to change the behavior.